### DISCUSSION TITLE
Function that run > 40 seconds

### DISCUSSION THREAD

**Discussion by sagar-jani** [vercel/community#2039] [Help] [answered]
### Summary

I have a requirement to run function for > 40 sec, I can't use edge func as it doesn't support langchain, pinecone etc 
Hobby tier only supports 10 seconds. 

What are the options free and paid ?

Have asked this question to Vercel on twitter but no answer!

### Additional information

I have a requirement to run function for > 40 sec, I can't use edge func as it doesn't support langchain, pinecone etc 
Hobby tier only supports 10 seconds. 

What are the options free and paid ?

Have asked this question to Vercel on twitter but no answer!

**Comment by amyegan** [comment, marked as answer]
Hi @sagar-jani! The serverless function timeout limit is currently 10 seconds for Hobby and 60 seconds for Pro. You can find more plan comparison information on the [pricing](https://vercel.com/pricing) page and in the [Limits & Limitations](https://vercel.com/docs/concepts/limits/overview#) documentation.

The team is also working to support more Node.js APIs, and a few were recently added. I'm not sure if these are enough to unblock you from using Edge Functions, which have a 30 second [initial response timeout](https://vercel.com/docs/concepts/functions/edge-functions/limitations#maximum-initial-response-time) across all plans, but you can learn more about the new expansion of Node.js compatibility in the changelog:  https://vercel.com/changelog/improved-node-js-compatibility-for-edge-middleware-and-edge-functions

**Reply by sagar-jani** [reply]
Thanks @amyegan , the edge functions currently don't support langchain, pinecone etc so I can't use that.
